glambie/util/timeseries_helpers.py

```diff
@@ -460,4 +486,84 @@ def resample_derivative_timeseries_to_monthly_grid(start_dates, end_dates, chang
     changes = resample_1d_array(dates, changes, monthly_grid)
     # 3 convert back to derivatives
     start_dates, end_dates, changes = cumulative_to_derivative(monthly_grid, changes, return_type="arrays")
-    return start_dates, end_dates, changes
+    if return_type == "arrays":
+        return start_dates, end_dates, changes
+    elif return_type == "dataframe":
+        return pd.DataFrame({"start_dates": start_dates, "end_dates": end_dates, "changes": changes})
+
+
+def get_total_trend(start_dates, end_dates, changes, return_type="dataframe"):
+    """
+    Calculates the full longterm trend of a derivatives timeseries
+
+    Parameters
+    ----------
+    start_dates : np.array or list of decimal dates
+        input start dates of each time period
+    end_dates : np.array or list of decimal dates
+        input end dates of each time period
+    changes : np.array or list
+        input changes between start and end date
+    return_type : str, optional
+        type in which the result is returned. Current options are: 'value' and 'dataframe', by default 'dataframe'
+
+    Returns
+    -------
+    pd.DataFrame or single value with overall trend
+        'dataframe': pd.DataFrame({'start_dates': start_dates, 'end_dates': end_dates, 'changes': changes})
+                     will contain a single row.
+        'value': longerm trend in input unit
+
+    """
+    if return_type == "value":
+        return np.nansum(changes)
+    elif return_type == "dataframe":
+        return pd.DataFrame({"start_dates": [np.nanmin(start_dates)],
+                             "end_dates": [np.nanmax(end_dates)],
+                             "changes": [np.nansum(changes)]})
+
+
+def get_average_trends_over_new_time_periods(start_dates, end_dates, changes, new_start_dates, new_end_dates):
```

Comment:
I see - I imagine you've used "trend" since this logic isn't necessarily for any one particular variable.

Reply:
yes, could also use average change I guess. but it's consistent with "get_total_trend"